modules/gapi/include/opencv2/gapi/s11n.hpp

```diff
@@ -142,7 +154,105 @@ namespace I {
+    GAPI_EXPORTS I::IStream& operator>> (I::IStream& is, cv::util::monostate &);
+
+    GAPI_EXPORTS I::OStream& operator<< (I::OStream& os, cv::GShape  shape);
+    GAPI_EXPORTS I::IStream& operator>> (I::IStream& is, cv::GShape &shape);
+
+    GAPI_EXPORTS I::OStream& operator<< (I::OStream& os, cv::detail::ArgKind  k);
+    GAPI_EXPORTS I::IStream& operator>> (I::IStream& is, cv::detail::ArgKind &k);
+
+    GAPI_EXPORTS I::OStream& operator<< (I::OStream& os, cv::detail::OpaqueKind  k);
+    GAPI_EXPORTS I::IStream& operator>> (I::IStream& is, cv::detail::OpaqueKind &k);
+
+    GAPI_EXPORTS I::OStream& operator<< (I::OStream& os, const cv::GArg &arg);
+    GAPI_EXPORTS I::IStream& operator>> (I::IStream& is,       cv::GArg &arg);
+
+
+    GAPI_EXPORTS I::OStream& operator<< (I::OStream& os, const cv::GKernel &k);
+    GAPI_EXPORTS I::IStream& operator>> (I::IStream& is,       cv::GKernel &k);
+
+    GAPI_EXPORTS I::OStream& operator<< (I::OStream& os, const cv::GMatDesc &d);
+    GAPI_EXPORTS I::IStream& operator>> (I::IStream& is,       cv::GMatDesc &d);
+
+    GAPI_EXPORTS I::OStream& operator<< (I::OStream& os, const cv::GScalarDesc &);
+    GAPI_EXPORTS I::IStream& operator>> (I::IStream& is,       cv::GScalarDesc &);
+
+    GAPI_EXPORTS I::OStream& operator<< (I::OStream& os, const cv::GOpaqueDesc &);
+    GAPI_EXPORTS I::IStream& operator>> (I::IStream& is,       cv::GOpaqueDesc &);
+
+    GAPI_EXPORTS I::OStream& operator<< (I::OStream& os, const cv::GArrayDesc &);
+    GAPI_EXPORTS I::IStream& operator>> (I::IStream& is,       cv::GArrayDesc &);
+
+    #if !defined(GAPI_STANDALONE)
+    GAPI_EXPORTS I::OStream& operator<< (I::OStream& os, const cv::UMat &);
+    GAPI_EXPORTS I::IStream& operator>> (I::IStream& is,       cv::UMat &);
+    #endif // !defined(GAPI_STANDALONE)
+
+    GAPI_EXPORTS I::OStream& operator<< (I::OStream& os, const cv::gapi::wip::IStreamSource::Ptr &);
+    GAPI_EXPORTS I::IStream& operator>> (I::IStream& is,       cv::gapi::wip::IStreamSource::Ptr &);
+
+    GAPI_EXPORTS I::OStream& operator<< (I::OStream& os, const cv::detail::VectorRef &);
+    GAPI_EXPORTS I::IStream& operator>> (I::IStream& is,       cv::detail::VectorRef &);
+
+    GAPI_EXPORTS I::OStream& operator<< (I::OStream& os, const cv::detail::OpaqueRef &);
+    GAPI_EXPORTS I::IStream& operator>> (I::IStream& is,       cv::detail::OpaqueRef &);
+
+    // FIXME: moved here due to clang lookup issue (see the comment above)
+    GAPI_EXPORTS I::OStream& operator<< (I::OStream& os, const cv::gimpl::RcDesc &rc);
+    GAPI_EXPORTS I::IStream& operator>> (I::IStream& is,       cv::gimpl::RcDesc &rc);
+
+    GAPI_EXPORTS I::OStream& operator<< (I::OStream& os, const cv::gimpl::Op &op);
+    GAPI_EXPORTS I::IStream& operator>> (I::IStream& is,       cv::gimpl::Op &op);
+
+    GAPI_EXPORTS I::OStream& operator<< (I::OStream& os, const cv::gimpl::Data &op);
+    GAPI_EXPORTS I::IStream& operator>> (I::IStream& is,       cv::gimpl::Data &op);
+
```

Comment:
This shouldn't be here as well. What user code requires these operators to be present right here??

Reply:
Removed